src/usr/local/www/diag_reboot.php

```diff
@@ -110,12 +121,33 @@ function startCountdown() {
 	</div>
 	<div class="panel-body">
 		<div class="content">
-			<p><?=gettext('Click "Reboot" to reboot the system immediately, or "Cancel" to go to the system dashboard without rebooting. (There will be a brief delay before the dashboard appears.)')?></p>
+			<p>
+<?php 
+if (!is_module_loaded("zfs.ko")) {
+	print('Click "Reboot" to reboot the system immediately, "FSCK Reboot" to reboot and run filesystem check, "Reroot" to stop processes, remount disks and re-run startup sequence or "Cancel" to go to the system dashboard without rebooting. (There will be a brief delay before the dashboard appears.)');
+	} else {
+	print('Click "Reboot" to reboot the system immediately, "FSCK Reboot" to reboot and run filesystem check or "Cancel" to go to the system dashboard without rebooting. (There will be a brief delay before the dashboard appears.)');
```

Comment:
There is a lot of shared/common/repeated text here that could be simplified. For example, if zfs is loaded, just add the reroot part at the end. It's also a bit crammed in there all run together.